transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -715,28 +717,56 @@ func convertVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, a
 
 }
 
-// converts a vertical index from one set of zoom parameters to another disregarding the floor() cacluation. This creates a simplier system of equations where the solu	tion set for height is a single variable. However, this does not describe the full solution set of height since we have excluded the floor calculation; it describes the condition where m = x, given m = floor(x) if and only if m <= x < m +1;
-func calculateMinVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, altitudeRangeScalar int64, indexOffset int64) (int64, error) {
+// converts a vertical index from one set of zoom parameters to another disregarding the floor() cacluation. This creates a simplier system of equations where the solution set for height is a single variable. However, this does not describe the full solution set of height since we have excluded the floor calculation; it describes the condition where m = x, given m = floor(x) if and only if m <= x < m +1;
+func calculateMinVerticalIndex(inputIndex int64, inputZoom int64, outputZoom int64, zBaseExponent int64, zBaseOffset int64) (int64, error) {
 
-	// check to make sure the input index exists in the input world
+	// 1. check that the input index exists in the input system
 	inputResolution := common.CalculateArithmeticShift(1, inputZoom)
 
 	maxInputIndex := inputResolution - 1
 	minInputIndex := -inputResolution
 
-	outputResolution := common.CalculateArithmeticShift(1, outputZoom)
-	maxOutputIndex := outputResolution - 1 + indexOffset
-	minOutputIndex := -outputResolution + indexOffset
-
 	if inputIndex > maxInputIndex || inputIndex < minInputIndex {
 		return 0, errors.NewSpatialIdError(errors.InputValueErrorCode, "input index does not exist")
 	}
 
-	// note that in the case of decimals, int64 rounds down to the nearest integer. This is desired behavior.
-	outputIndex := (indexOffset) + common.CalculateArithmeticShift(inputIndex, (outputZoom-inputZoom+altitudeRangeScalar))
+	// 2. Calculate outputIndex
+	// note: outputIndex = alphaTerm + indexTerm
+
+	// 2.1: Create the alpha term, which is is a special case of the arithmatic shift: it should always "round" towards 0.
+
+	// unitScalar determines the appropriate positive or negative sign on zed. unitScalar is 1 if zBaseOffset is positive (>=0)
+	// and unitScalar is negative if zBaseOffset is negative (<0). This is necessary for using CalculateArithmeticShift() on
+	// zBaseOffset that results in the desired behavior. CalculateArithmeticShift always rounds down to the next lowest integer,
+	// even when the input is negative. However, in the case of the alpha term, we should always "round towards 0".
```

Comment:
No, we must not round towards 0 but down.
Otherwise, if we set zBaseOffset as -1, outputZoom as 24 and zBaseExponent as 25, alphaTerm will be not -1 but 0.
This is wrong.

Reply:
In this example, (zBaseExponent=25, zoomLevel=25, where the resulting voxelHeight=1m and shift distance is equivalent to 1m) the one-index shift in zBaseOffset terms is not enough magnitude to equal an entire one-index shift in the output terms (zBaseExponent= 25, outputZoom=24, where the resulting voxelHeight=2m). This holds true if zBaseOffset is either -1 or 1. Therefore, the alphaTerm should be 0. 

note: [] denotes the use of CalculateArithmeticShift() function. The result of this function always rounds down to the nearest integer.
{} denotes the use of the altered Arithmetic shift function which rounds towards zero.

```math
alphaTerm = \{zBaseOffset*2^{outputZoom-zBaseOffset} \} 
```

### Example 1:
zBaseOffset=1 (equivalent to a shift of 1m), zBaseExponent=25, outputZoom=24, where the resulting voxelHeight=2m

```
alphaTerm = {1*2^(24-25)}
alphaTerm = {1*0.5}
alphaTerm = 0
```

Intuitively, the alphaTerm should be 0, because a zBaseOffset shift of 1 does not shift enough to result in a one-index shift in the output (where one index = 2m )

### Example 2:
zBaseOffset= -1 (equivalent to a shift of -1m), zBaseExponent=25, outputZoom=24, where the resulting voxelHeight=2m

```
alphaTerm = {-1*2^(24-25)}
alphaTerm = {-1*0.5}
alphaTerm = 0
```

Intuitively, the alphaTerm should be still be 0, because, just as in example 1, a zBaseOffset shift of -1 does not shift enough to result in a one-index shift down in the output (where one index = 2m )

### Example 3:
zBaseOffset= -1 (equivalent to a shift of -1m), zBaseExponent=25, outputZoom=24, where the resulting voxelHeight=2m

```
(here we "round down" instead )
alphaTerm = [-1*2^(24-25)]
alphaTerm = [-1*0.5]
alphaTerm =  -1
```

Here, it doesn't make sense to shift the index in the output 1 unit down: the distance shifted is only half an index distance in the output. This especially doesn't make sense when we consider that a positive shift of 1 (zBaseOffset=1) as seen in example 1 results in 0, and a shift down (-minus) of the same magnitude would result in -1. The magnitude of zBaseOffset should be the same whether the sign is positive or negative. Therefore, the must round to the integer closest to 0.

Reply:
No, this is not a problem of intuition but of consistency.
Example 3 makes sense.
The magnitude of zBaseOffset should not be the same whether the sign is positive or negative.
We must round it down.
I'll prove it.

Let's think about the extension of Example 1 and Example 2.
We assume zBaseExponent=25 and outputZoom=24.
You showed the following exmples.
| zBaseOffset | alphaTerm |
| ---- | ---- |
| 1 | 0 |
| -1 | 0 |

Then, I extend this to the following examples.
| zBaseOffset | alphaTerm |
| ---- | ---- |
| 7 | 3 |
| 6 | 3 |
| 5 | 2 |
| 4 | 2 |
| 3 | 1 |
| 2 | 1 |
| 1 | 0 |
| 0 | 0 |
| -1 | 0 |
| -2 | -1 |
| -3 | -1 |
| -4 | -2 |
| -5 | -2 |
| -6 | -3 |

This shows that alphaTerms other than 0 have two zBaseOffsets, but only one alphaTerm 0 has three zBaseOffsets!
This is inconsistent.

Therefore, we must round it down as the following exmples.
| zBaseOffset | alphaTerm |
| ---- | ---- |
| 7 | 3 |
| 6 | 3 |
| 5 | 2 |
| 4 | 2 |
| 3 | 1 |
| 2 | 1 |
| 1 | 0 |
| 0 | 0 |
| -1 | -1 |
| -2 | -1 |
| -3 | -2 |
| -4 | -2 |
| -5 | -3 |
| -6 | -3 |

Thus, all alphaTerms have two zBaseOffsets consistently.
Do not make 0 special.

Reply:
In discussion with our team, it was decided that rounding towards zero achieves the desired outcome. This would mean that [Example 3](https://github.com/trajectoryjp/spatial_id_go/pull/24#discussion_r1611244766) correctly describes a case of incorrect rounding and that the "round towards zero" method will be applied.

Reply:
No, our team does not decide it.